BASE/src/ServerTelemetryChannel/Implementation/TelemetrySerializer.cs

```diff
@@ -27,6 +29,33 @@ public Uri EndpointAddress
         }
 
         public virtual void Serialize(ICollection<ITelemetry> items)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items);
+            this.Transmitter.Enqueue(transmission);
+        }
+
+        public virtual async Task<bool> Serialize(ICollection<ITelemetry> items, bool manualFlushAsync)
+        {
+            this.HandleTelemetryException(items);
+
+            var transmission = new Transmission(this.EndpointAddress, items, manualFlushAsync);
+            
+            using (var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(100)))
```

Comment:
I would recommend adding a `CancellationToken` parameter to the method (i.e. have the caller decide when to cancel).